File: HomeWork/Home4.1/Pizza.java
import java.util.ArrayList;
import java.util.List;

public class Pizza {
    private List<Ingredient> ingredients = new ArrayList<>();

    public void addIngredient(String name, double cost) {
        ingredients.add(new Ingredient(name, cost));
    }

    public double getTotalCost() {
        return ingredients.stream().mapToDouble(Ingredient::getCost).sum();
    }

    public List<Ingredient> getIngredients() {
        return ingredients;
    }

    public static class Ingredient {
        private final String name;
        private final double cost;

        public Ingredient(String name, double cost) {
            this.name = name;
            this.cost = cost;
        }

        public String getName() {
            return name;
        }

        public double getCost() {
            return cost;
        }
    }
}

File: Class/practice4.1/Pizza.cpp
#include "Pizza.h"

void Pizza::addIngredient(const std::string& ingredient, double cost) {
    ingredients.emplace_back(ingredient, cost);
}

double Pizza::getTotalCost() const {
    double total = 0.0;
    for (const auto& ingredient : ingredients) {
        total += ingredient.second;
    }
    return total;
}

std::vector<std::pair<std::string, double>> Pizza::getIngredients() const {
    return ingredients;
}
